charts/prometheus-alerts/templates/containers-prometheusrule.yaml

```diff
@@ -480,4 +480,166 @@ spec:
         {{- end }}
     {{- end }}
 
+  - name: {{ .Release.Name }}.{{ .Release.Namespace }}.alertRulesSelectorsValidity
```

Comment:
Ok. This chart maybe needs a little more reworking than this. I think that each of the core types (DaemonSets, Deployments, StatefulSets, Jos, HPAs) should have its own settings section. Each of those sections should be controlled with an on-off flag, and it should turn off ALL the alerts.

Eg:

```yaml
containerRules:
  enabled: true

  pods:
    enabled: true
    PodContainerTerminated:
      severity: warning
      threshold: 0
      over: 10m
      for: 1m
      reasons:
        # - Error  < when a container is evicted gracefully, the "error" state is used.
        - ContainerCannotRun
        - DeadlineExceeded

  deployments:
    enabled: true
    KubeDeploymentGenerationMismatch:
      severity: warning
      for: 15m

  statefulSets:
    enabled: true
    KubeStatefulSetReplicasMismatch:
      severity: warning
      for: 15m
```

Does that make sense? This would obviously be a breaking change - so we probably would add in a safety check that errors out if it finds old values (`.Values.containerRules.PodContainerTerminated`, etc..). Alternatively, we could go as far as implementing a schema file ... but that might be just a pain in the ass.

Reply:
yeah, ok i was thinking re-writing the entire chart as well... I like the safety check idea. That was my main concern there with re-writing.

Reply:
I think this is going to be a huge change so let me restructure first in 1 PR and then add the new alerts back in a second follow up PR.

Reply:
https://github.com/Nextdoor/k8s-charts/pull/295